### ISSUE TITLE
Running inference on engine based on onnx model: Error Code 1: Myelin ([myelinGraphExecute] Called without resolved dynamic shapes

### ISSUE THREAD

**Issue by ece85** [NVIDIA/TensorRT#2919]
## Description

I am trying to create an object detection engine that outputs image feature embeddings with detections.  I have successfully done this with onnx 1.13.1 with Tensorrt 8.0.  I am unable to do this with newer versions of Tensorrt (8.5.x and 8.6.x).

With my latest environment (described below), I can successfully create a engine with no Errors (via trtexec --onnx option), but when i try to run inference with trt in python, i get the following error:

[04/28/2023-23:25:38] [TRT] [E] 1: [runner.cpp::executeMyelinGraph::715] Error Code 1: Myelin ([myelinGraphExecute] Called without resolved dynamic shapes.)

The outputs that are returned seem to have all zero (or near zero values).
![image](https://user-images.githubusercontent.com/4247956/235269420-43c6924f-e61a-4051-bc4f-253250232d75.png)

One thing that is interesting to note.  When i run this with onnx 1.13.1 and Tensorrt 8.0, Tensorrt Interprets my onnx file as having an output of dimension of 100x8  (100 detections where each detection is an array of length 8).  However with newer versions of TenosrRT the model is interpreted as having an output dimension of -1x8.  I think this is the root of the issue.

## Environment

**TensorRT Version**: 8.6.1
**Onnx Version**: 1.12.0
**NVIDIA GPU**: 3080ti
**NVIDIA Driver Version**: nvidia-driver-530 
**CUDA Version**: 12.1
**CUDNN Version**: 8.9.0.131-1+cuda12.1
**Operating System**: ubuntu 18.04.6 LTS
**Python Version (if applicable)**: 3.8.10
**Tensorflow Version (if applicable)**: N/A
**PyTorch Version (if applicable)**:  2.0.0+cu118
**Baremetal or Container (if so, version)**:  Docker Image: nvcr.io/nvidia/tensorrt:23.04-py3

## Relevant File

repo: 

https://drive.google.com/file/d/1pF_qXym4zuYUPp5ByeJL9wt8viuWAWe0/view?usp=share_link

the onnx file is in the zipfile in /bugReport/scripts/best_w_embeddings.onnx

## Steps To Reproduce
unzip repo.

cd /bugReport/docker
./build_docker.sh
./run_docker_dev.sh
python3 test_onnx_trt.py 

test_onnx_trt.py  converts to trt engine via trtexec call, then deserializes plan, and processes an image.  it should log an error:

[04/28/2023-23:58:27] [TRT] [E] 1: [runner.cpp::executeMyelinGraph::715] Error Code 1: Myelin ([myelinGraphExecute] Called without resolved dynamic shapes.)

**Comment by zerollzeng** [comment]
The issue is in your inference scripts or your environment.

if you can build the engine with trtexec, after building the engine trtexec will launch inference for perf summary, so the engine it self is fine. you can double confirm this by `trtexec --loadEngine=engine.plan`

I also check the model with `polygraphy run --trt --onnxrt best_w_embeddings.onnx`

I can see it passed and the output is match with onnx runtime.

**Comment by ece85** [comment]
@zerollzeng thank you for your time.

I get a failed test output comparison when running the following command from within the docker container that was linked in this issue report:

`polygraphy run --trt --onnxrt best_w_embeddings.onnx`

following is the last few lines of the report:
` Comparing Output: 'output' (dtype=float32, shape=(0, 8)) with 'output' (dtype=float32, shape=(0, 8))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         trt-runner-N0-05/03/23-20:22:41: output | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at 0, max=0 at 0, avg-magnitude=0
[I]         onnxrt-runner-N0-05/03/23-20:22:41: output | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at 0, max=0 at 0, avg-magnitude=0
[I]         Error Metrics: output
[I]             Minimum Required Tolerance: elemwise error | [abs=0] OR [rel=0] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at 0, max=0 at 0, avg-magnitude=0
[I]             Relative Difference | Stats: mean=0, std-dev=0, var=0, median=0, min=0 at 0, max=0 at 0, avg-magnitude=0
[I]         PASSED | Output: 'output' | Difference is within tolerance (rel=1e-05, abs=1e-05)
[E]     FAILED | Mismatched outputs: ['p3', 'p4', 'p5']
[E] Accuracy Summary | trt-runner-N0-05/03/23-20:22:41 vs. onnxrt-runner-N0-05/03/2
[polygraphy_trt_onnxrt_best_w_embeddings.onnx.txt]
3-20:22:41 | Passed: 0/1 iterations | Pass Rate: 0.0%
[E] FAILED | Runtime: 146.521s | Command: /usr/local/bin/polygraphy run --trt --onnxrt ./best_w_embeddings.onnx`

complete log: 
[polygraphy_trt_onnxrt_best_w_embeddings.onnx.txt](https://github.com/NVIDIA/TensorRT/files/11391041/polygraphy_trt_onnxrt_best_w_embeddings.onnx.txt)


I think the fundamental difference i see between tensorrt 8.0 and 8.6 (and 8.5) is the size of the output node named 'output' as they come out of the python libarary tensorrt.OnnxParser() function. 
I am using the following script to generate a engine from an onxx file:  https://github.com/Linaom1214/tensorrt-python/blob/2223039fca794932d9aedef0007b2fbd9dae0a8e/export.py

code snippet from that script:
![image](https://user-images.githubusercontent.com/4247956/236043087-a20b47f0-89d5-4aee-935e-10f088f3dc3f.png)


when the output is logged via line 143, my output node named 'output' is of size 100x8 using Tensorrt 8.0, it is -1x8 when using newer versions (ive tested 8.5 and 8.6)

**Comment by zerollzeng** [comment]
polygraphy has strict diff comparison by default, if you look at the `Absolute Difference` for each output, you will found it's very small, like 10e-5.
